packages/pwa-kit-react-sdk/src/ssr/server/react-rendering.js

```diff
@@ -84,23 +87,27 @@ const initAppState = async ({App, component, match, route, req, res, location})
     const {params} = match
 
     const components = [App, route.component]
-    const promises = components.map((c) =>
-        c.getProps
-            ? c.getProps({
-                  req,
-                  res,
-                  params,
-                  location
-              })
-            : Promise.resolve({})
-    )
+    const queries = queryClient.queryCache.queries.map((q) => q.fetch())
```

Comment:
what happens when the query throws? Is react-query able to hydrate the error?

Reply:
There are some [limitations](https://tanstack.com/query/v4/docs/reference/hydration#limitations) with serialization of errors. I can play around with it to see the actual behaviour, because it's not completely clear what happens to errors if you tell dehydrate to serialize them. Since it says you can't do that.

Reply:
I see. From the documentation, it looks like react-query does not provide a serializer for errors.

Reply:
It would be likely that the error would be handled one of a handful of ways that don't result in the need for serializing the error. 

1. The error is thrown and our server pipeline will do the appropriate things with it, e.g. set the status code, render the error page.
2. The error is not thrown and handled on the page in which the page would not render an appropriate UI. 

I still want to see what would happen is a query throws an error internally in its query function, then on the client we wouldn't re-run that query. I'm curious as to what data and or error are set to.